pages/article/ArticleHandler.inc.php

```diff
@@ -57,11 +69,24 @@ function initialize($request, $args) {
 		$articleId = isset($args[0]) ? $args[0] : 0;
 		$galleyId = isset($args[1]) ? $args[1] : 0;
 
-		$journal = $request->getContext();
-		$publishedArticleDao = DAORegistry::getDAO('PublishedArticleDAO');
-		$publishedArticle = $publishedArticleDao->getPublishedArticleByBestArticleId((int) $journal->getId(), $articleId, true);
+		$this->journal = $request->getContext();
 
 		$issueDao = DAORegistry::getDAO('IssueDAO');
+		$articleDao = DAORegistry::getDAO('ArticleDAO');
+		$publishedArticleDao = DAORegistry::getDAO('PublishedArticleDAO');
+
+		// get all published previous article versions
+		$this->previousRevisions = $articleDao->getPublishedSubmissionRevisions($articleId);
```

Comment:
should we not get here just submission revisions instead of published submission revisions? -- because it could be that the editor previews a not published article and because you search the published articles a few lines below